### ISSUE TITLE
Add Azure DevOps CI/CD repository and pipeline support option

### ISSUE THREAD

**Issue by obriensystems** [terraform-google-modules/terraform-example-foundation#1205]
### TL;DR

assigned to https://github.com/fmichaelobrien
PR developed in fork https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/issues/399

## ER:
The default CI/CD repo/pipeline around Cloud Build and Cloud Source Repositories works well.
There is a need for Azure DevOps (ADO) integration as a CI/CD option.
A large proportion of our Canadian Public Sector clients (provincial, federal, crown corporation) use ADO as their repository and build system.
- https://learn.microsoft.com/en-us/azure/devops/organizations/billing/overview?view=azure-devops
- 

## Work:
similar to our github/gitlab/terraform-cloud and jenkins support on top of default CB/CSR support - add versions of the following artifacts

- specific agent requirements and terraform module like
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/0-bootstrap/modules/jenkins-agent/README.md#Requirements
- specific 0-bootstrap and subsequent READMEs like
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/0-bootstrap/README-Jenkins.md
- section in the root README
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/README.md#0-bootstrap
- ADO pipeline yaml for plan and apply like
- https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/build/gitlab-ci.yml

**Comment by fmichaelobrien** [comment]
Test terraform 1.3.10 image in Azure Devops

<img width="1573" alt="Screenshot 2024-04-27 at 18 49 14" src="https://github.com/GoogleCloudPlatform/pbmm-on-gcp-onboarding/assets/24765473/f2f14ac6-a119-4fb4-bbb3-0c651a19a526">

via azure-pipeline.yml
```
trigger:
- main

pool:
  vmImage: ubuntu-latest
- script: |
    docker run obrienlabs/terraform-example-foundation-ado --version 
```
